### PR TITLE
gles3 initial versions

### PR DESCRIPTION
**Pull request by craff**
I restarted from a fresh fork ... and now have a clean patch (I hope)

### CODE REVIEW

**Comment by craff** [comment]
I see that some test fails ... but I can't access any details ... How can I know what's happening ,

**Comment by dsheets** [comment]
I think if you click `Show all checks` and then `Details` next to `continuous-integration/travis-ci/pr`, you will be taken to https://travis-ci.org/ocaml/opam-repository/builds/114206682 which shows all the test runs.

On Linux:

```
### stderr ###
Found params: num = 156, size = 313, mask = 0x43A
# Probes: max = 3, sum = 42
# ml_gles3.c:31:23: fatal error: GLES3/gl3.h: No such file or directory
# compilation terminated.
# make: *** [ml_gles3.o] Error 1
```

On OS X:

```
### stderr ###
Found params: num = 156, size = 313, mask = 0x43A
# Probes: max = 3, sum = 42
# ml_gles3.c:25:10: fatal error: 'caml/misc.h' file not found
# #include <caml/misc.h>
#          ^
# 1 error generated.
# make: *** [ml_gles3.o] Error 1
```

File: packages/gles3/gles3.20160307.alpha/opam [hunk 1]
```diff
@@ -0,0 +1,14 @@
+opam-version: "1.2"
+maintainer: "Christophe Raffalli <raffalli@univ-savoie.fr>"
+authors:
+  "Christophe Raffalli <raffalli@univ-savoie.fr>, Alexandre Miquel<amiquel@fing.edu.uy>"
+homepage: "http://lama.univ-savoie.fr/~raffalli/gles3"
+bug-reports: "raffalli@univ-savoie.fr"
+license: "LGPL"
```
Comment (dsheets) [inline packages/gles3/gles3.20160307.alpha/opam]: Could you please specify the version of the LGPL license and whether you intend to use the "OCaml linking exception"?

**Comment by dsheets** [comment]
Please see the other comments on #5893. Also, you can simply force-push to a branch that is the source of a pull request to update it instead of creating a new branch and a new pull request. This helps to keep the conversation about the PR in one place and reduce the volunteer effort required to review submissions.

**Comment by craff** [comment]
I did not see all comment, sorry.
I fixed the LGPL ...
However, I do not understand this comment:

packages/gles3/gles3.20160307.alpha/url
@@ -0,0 +1,3 @@
+http:
+  "https://lama.univ-savoie.fr/~raffalli/gles3/gles3-20160307.alpha.tar.gz"
@dsheets OCaml member
dsheets added a note 4 hours ago

I believe this can be archive: in order to maintain flexibility in transport.

**Comment by dsheets** [comment]
It is very minor but `http:` (opam field) specifies the transport protocol which is already included in the URL as `https:` (URI scheme). See https://opam.ocaml.org/doc/manual/dev-manual.html#sec10 which discusses the various `url` file fields and their meanings. `archive` will use the URL whereas `http` is a repetition of the type declaration and becomes another component to update if the archive location changes how it is hosted.

With all that said, lots of other packages use `http:` where `archive:` would be more appropriate so it's not very important.

**Comment by dsheets** [comment]
... and I see that this is probably an opam-publish issue. Sorry to bother you, @craff.

**Comment by craff** [comment]
I have stopped to use opam-publish ... to me it should just git commit to your own fork of the opam-repository ... The pull request can be manual ... My changes are coming soon

**Comment by craff** [comment]
### stderr ###
Found params: num = 156, size = 313, mask = 0x43A
# Probes: max = 3, sum = 42
# ml_gles3.c:31:23: fatal error: GLES3/gl3.h: No such file or directory
# compilation terminated.
# make: *** [ml_gles3.o] Error 1


This is normal if GLES3 is not installed on the machine ?

**Comment by craff** [comment]
### stderr ###
Found params: num = 156, size = 313, mask = 0x43A
# Probes: max = 3, sum = 42
# ml_gles3.c:25:10: fatal error: 'caml/misc.h' file not found
# #include <caml/misc.h>
#          ^
# 1 error generated.
# make: *** [ml_gles3.o] Error 1

This looks to me like a wrong ocaml installation ... moreover, I do not expect gles3 to work on OSX so easily ?

**Comment by OCamlPro-Henry** [comment]
For the missing header, what we usually do is a second "configuration" package, that checks the header avaibility and provides information about the corresponding "system package" on usual distros. See for instance: 

  https://github.com/ocaml/opam-repository/blob/master/packages/conf-cairo/conf-cairo.1/opam

In particular, the `depext` field is used by the CI to install the required system package.

**Comment by OCamlPro-Henry** [comment]
If the library is not supposed to compile on OS X, you might add `available: [ os != "darwin" ]` in the OPAM package description.

**Comment by craff** [comment]
I think the package should be ok ... except that I don't know how to produce the depexts for all distribution on earth ;-)

**Comment by OCamlPro-Henry** [comment]
You already added quite a few distro in `depexts`, thanks :-)

Unfortunatly, the CI failed because the header file `GLES/gl3.h` is not available on Ubuntu/Precise (which is used by the CI). It only appeared with Trusty. I am not aware of any mechanism in OPAM to handle that, so the package looks good to me. I am just waiting for someone to confirm that before merging, @dsheets ?

Thanks again for submitting the package !

**Comment by dsheets** [comment]
@OCamlPro-Henry you are correct. LGTM.

**Comment by dsheets** [comment]
Oh, one last thing, it might be nice to include a `post-messages` field in the failure condition to explain possible reasons. See <https://opam.ocaml.org/doc/manual/dev-manual.html#sec9> search for `post-messages`.

**Comment by dsheets** [comment]
I've open AltGr/Camelus#7 to track the `depext`/`post-messages` best practice policy.

**Comment by craff** [comment]
I added post-messages. We can not ask a minimal version in depexts ? This would make 
gles3 unavailable under ubuntu 12.04 or before ?

**Comment by dsheets** [comment]
Unfortunately, we don't currently have versioning incorporated into the depext system due to complexity of the interface task. Some level of this kind of functionality is certainly possible and I have open ocaml/opam-depext#43 to track that.

**Comment by craff** [comment]
s/or/os/ done

**Comment by craff** [comment]
By the way, the typo s/or/os did not give an error when I installed the package from by own
git repository ... Why ?

**Comment by dsheets** [comment]
No error is generated because `or` is a valid (but unbound) variable name. I've opened AltGr/Camelus#8 for this.

**Comment by craff** [comment]
Is the package OK now or do you have more suggestion ?

**Comment by craff** [comment]
Thanks !

**Comment by OCamlPro-Henry** [comment]
LGTM, thanks for introducing the `post-message` !

**Comment by craff** [comment]
And thanks again for all the help with opam ...

As an aside question: opam2debian is still alive ? It is not in the opam repository. I think tools like opam2x
where x would range in the depexts of the package would be a must ... This tool could even use opam
for build and/or install the package of some distros.

One could even imagine that all opam package automatically become package for 
all distro (except if we do not want that).

**Pull request merged**